lib/public/js/docs-nav.js

```diff
@@ -1,41 +1,140 @@
 'use strict';
 /* global document */
 
-const Navbar = require('navbar');
+const ScrollIntoViewIfNeeded = require('scroll-into-view-if-needed');
+const Navbar = require('./vendor/navbar');
 
 const internals = {};
 
+const FIXED_HEADER_HEIGHT = 80;
+const TOP_PADDING = 35;
+
 exports.init = (contentEl, navEl) => {
 
     if (!contentEl) {
         return;
     }
 
     if (!navEl) {
-        throw new Error('Docs navigation content element exists, but not the nav elements.');
+        throw new Error('Docs navigation content element exists, but not the nav element.');
     }
 
+    const hashState = new internals.HashState();
+
     const navbar = Navbar({
         tagName: navEl.tagName,
         elementList: contentEl.querySelectorAll('h1, h2, h3, h4, h5, h6'),
         makeNavListItem: (el) => {
 
-            const text = el.textContent.trim();
+            el = el.cloneNode(true);
+
+            const anchor = el.removeChild(el.querySelector('a'));
+            const indent = Number(el.tagName.replace('H', ''));
 
-            const item = internals.render(
-                internals.item(text, el.tagName.replace('H', 'indent-'))
+            return internals.render(
+                internals.item(el.innerHTML, anchor.hash, indent)
             );
+        },
+        onScroll: (navItem) => {
 
-            item.addEventListener('click', () => el.scrollIntoView({ block: 'start', behavior: 'smooth' }));
+            if (window.location.hash !== navItem.hash && !hashState.autoScrolling && !hashState.firstScroll) {
```

Comment:
fun stuff ;)

Reply:
Hahaha, ahhhh I can't wait to clean this up!